apps/functions-subscription/api/internal.yaml

```diff
@@ -0,0 +1,159 @@
+openapi: 3.0.1
+info:
+  version: 0.0.0
+  title: Trial System API
+  description: |
+    Documentation of the Trial System Subscription function.
+servers:
+  - url: https://ts-p-itn-subscription-fn-01.azurewebsites.net
+security:
+  - ApiKeyAuth: []
+paths:
+  /info:
+    get:
+      operationId: info
+      responses:
+        '200':
+          $ref: '#/components/responses/200Info'
+        '404':
+          $ref: '#/components/responses/404NotFound'
+        '500':
+          $ref: '#/components/responses/500InternalServerError'
+  /trials/{trialId}/subscriptions/{userId}:
+    get:
+      operationId: getSubscription
+      summary: Get subscription detail
+      description: >-
+        Returns the details about the subscription of the specified user for
+        the specified trial.
+      parameters:
+        - $ref: '#/components/parameters/pathTrialId'
+        - $ref: '#/components/parameters/pathUserId'
+      responses:
+        '200':
+          $ref: '#/components/responses/200Subscription'
+        '401':
+          $ref: '#/components/responses/401Unauthorized'
+        '404':
+          $ref: '#/components/responses/404NotFound'
+        '500':
+          $ref: '#/components/responses/500InternalServerError'
+  /trials/{trialId}/subscriptions:
+    post:
+      operationId: createSubscription
+      summary: Create a subscription
+      description: Subscribe the given user to the given trial.
+      parameters:
+        - $ref: '#/components/parameters/pathTrialId'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateSubscription'
+      responses:
+        '200':
```

Comment:
**thought**: what do you think about the `201` status code?

IMHO if the resource has been created, then we could explicit that using the [`201 Created` status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201).

Otherwise, if we don't know if the resource has been created (or if there will be something else, maybe async, that will process the request, we could use the [`202 Accepted` status code](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/202).

What are you thoughts about this?

Reply:
I agree 👍 , I'm going to change it

Reply:
agree :)

Reply:
Done ✅

Reply:
Done ✅